internal/resource/http.go

```diff
@@ -109,27 +109,45 @@ func (f *Fetcher) UpdateHttpTimeoutsAndCAs(timeouts types.Timeouts, cas []types.
 		if err != nil {
 			return err
 		}
-		block, _ := pem.Decode(cablob)
-		if block == nil {
-			f.Logger.Err("Unable to decode CA (%v)", ca.Source)
-			return ErrPEMDecodeFailed
-		}
-
-		cert, err := x509.ParseCertificate(block.Bytes)
+		// For Root CA
+		restCAs, err := f.parseCABundle(cablob, ca, pool)
 		if err != nil {
-			f.Logger.Err("Unable to parse CA (%v): %s", ca.Source, err)
+			f.Logger.Err("Unable to parse CA bundle: %s", err)
 			return err
 		}
-
-		f.Logger.Info("Adding %q to list of CAs", cert.Subject.CommonName)
-		pool.AddCert(cert)
+		// For Intermediate CAs
+		for len(restCAs) > 0 {
+			res, err := f.parseCABundle(restCAs, ca, pool)
+			if err != nil {
+				f.Logger.Err("Unable to parse CA bundle: %s", err)
+				return err
+			}
+			restCAs = res
+		}
 	}
-
 	f.client.transport.TLSClientConfig = &tls.Config{RootCAs: pool}
-
 	return nil
 }
 
+// parseCABundle parses a caBundle which includes a Root CA and other
+// intermediate CAs.
+func (f *Fetcher) parseCABundle(cablob []byte, ca types.Resource, pool *x509.CertPool) ([]byte, error) {
+	block, remainingCAs := pem.Decode(cablob)
+	if block == nil {
+		f.Logger.Err("Unable to decode CA (%v)", ca.Source)
+		return nil, ErrPEMDecodeFailed
+	}
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		f.Logger.Err("Unable to parse CA (%v): %s", ca.Source, err)
+		return nil, err
+	}
+	f.Logger.Info("Adding %q to list of CAs", cert.Subject.CommonName)
+	pool.AddCert(cert)
+
+	return remainingCAs, nil
```

Comment:
Returning the remaining bytes seems odd.  I'd be inclined to put the loop in this function, not in the caller.